eng/scripts/Language-Settings.ps1

```diff
@@ -29,7 +29,7 @@ function Get-AllPackageInfoFromRepo([string]$serviceDirectory = $null) {
   if ($serviceDirectory) {
     $searchPath = Join-Path $sdkRoot $serviceDirectory
     Write-Host "searchPath=$searchPath"
-    [array]$ymlFiles = Get-ChildItem -Path $searchPath "ci*.yml" | Where-Object { $_.PSIsContainer -eq $false}
+    [array]$ymlFiles = Get-ChildItem -Recurse -Path $searchPath "ci*.yml" | Where-Object { $_.PSIsContainer -eq $false}
```

Comment:
We had the reverse issue in the go repo, where there were nested ci.yml files that weren't intended to be sucked up by parent level pipelines. Seems there are a couple cases, like with maps, where there is both a `sdk/maps/ci.yml` file AND a `sdk/maps/<sub dir>/ci.yml` file. Not sure if this will cause unintended pipeline consequences there vs. what's intended in communication. Just FYI

```
./sdk/maps/azure-maps-geolocation/ci.yml
./sdk/maps/azure-maps-route/ci.yml
./sdk/maps/azure-maps-traffic/ci.yml
./sdk/maps/azure-maps-render/ci.yml
./sdk/maps/azure-maps-timezone/ci.yml
./sdk/maps/azure-maps-weather/ci.yml
./sdk/maps/azure-maps-search/ci.yml
./sdk/spring/spring-cloud-azure-starter-monitor/ci.yml
./sdk/communication/azure-communication-callingserver/ci.yml
./sdk/communication/azure-communication-sms/ci.yml
./sdk/communication/azure-communication-callautomation/ci.yml
./sdk/communication/azure-resourcemanager-communication/ci.yml
./sdk/communication/azure-communication-chat/ci.yml
./sdk/communication/azure-communication-rooms/ci.yml
./sdk/communication/azure-communication-email/ci.yml
./sdk/communication/azure-communication-messages/ci.yml
./sdk/communication/azure-communication-common/ci.yml
./sdk/communication/azure-communication-jobrouter/ci.yml
./sdk/communication/azure-communication-identity/ci.yml
./sdk/communication/azure-communication-phonenumbers/ci.yml
./sdk/identity/azure-identity-extensions/ci.yml
./sdk/eventgrid/azure-resourcemanager-eventgrid/ci.yml
```

Reply:
@benbp you're absolutely right and I'm going to wait to merge this till @JimSuplizio returns to merge this PR. Nothing is super urgent, I'm just trying to help @mssfang

Reply:
Yeah. Thank you for the help @scbedd It unblocks my release pipeline. No longer see those occurred errors.

Reply:
@scbedd and @mssfang, this isn't the right fix. When [I wrote this function a few weeks back](https://github.com/Azure/azure-sdk-for-java/pull/42907/files#diff-1d28dc86baa237ca6efe6224431142a9e8bbff66fa10a2c20c69031a22f5b387R41), the lack of the `-Recurse` in the command was intentional. If -Recurse is added, it'll break things, and we'll end up with exact problem I'd just fixed which was packageInfo generation was too greedy. In this particular case, if `-Recurse` was used it would get all of the communication libraries' packageInfo files, including ones that weren't release.